server/master/src/main/java/org/apache/accumulo/master/tableOps/MoveExportedFiles.java

```diff
@@ -48,21 +55,35 @@
 
       Map<String,String> fileNameMappings = PopulateMetadataTable.readMappingFile(fs, tableInfo);
 
-      for (String oldFileName : fileNameMappings.keySet()) {
-        if (!fs.exists(new Path(tableInfo.exportDir, oldFileName))) {
-          throw new AcceptableThriftTableOperationException(tableInfo.tableId, tableInfo.tableName,
-              TableOperation.IMPORT, TableOperationExceptionType.OTHER,
-              "File referenced by exported table does not exists " + oldFileName);
-        }
-      }
+      FileStatus[] exportedFiles = fs.listStatus(new Path(tableInfo.exportDir));
+      FileStatus[] importedFiles = fs.listStatus(new Path(tableInfo.importDir));
+
+      Set<String> importing = Arrays.stream(exportedFiles).map(fstat -> fstat.getPath().getName())
+        .map(fileNameMappings::get).collect(Collectors.toSet());
 
-      FileStatus[] files = fs.listStatus(new Path(tableInfo.exportDir));
+      Set<String> imported = Arrays.stream(importedFiles).map(fstat -> fstat.getPath().getName())
+        .collect(Collectors.toSet());
+      log.debug("Files already present in imported (target) directory: {}",
+        StringUtils.join(imported.toArray(), ","));
 
-      for (FileStatus fileStatus : files) {
+      Set<String> missingFiles = Sets.difference(new HashSet<String>(fileNameMappings.values()),
+        new HashSet<String>(Sets.union(importing, imported)));
+
+      if (!missingFiles.isEmpty())
+      {
```

Comment:
This doesn't match our formatter style. Before we merge this, make sure the code is formatted and the basic QA checks pass (`mvn clean verify -DskipTests`).

Reply:
For small changes I only `mvn compile` to format the code and let Travis CI do the basic quality checks.

Reply:
That's fine, too. I typically do it locally, since it doesn't take that long, and it results in fewer round trips to resolve all the issues. Also, we have limited executors on Travis that we share with the rest of the ASF, so it can take awhile before it runs in Travis.

Reply:
Sorry for the overlook. I will ensure the QA checks pass before the next commit.

Reply:
Addressed in latest commit